pkgs/build-support/rust/default.nix

```
@@ -119,7 +120,8 @@ stdenv.mkDerivation (args // {
   # After unpacking and applying patches, check that the Cargo.lock matches our
   # src package. Note that we do this after the patchPhase, because the
   # patchPhase may create the Cargo.lock if upstream has not shipped one.
-  postPatch = (args.postPatch or "") + stdenv.lib.optionalString validateCargoDeps ''
+  postPatch = stdenv.lib.optionalString validateCargoDeps ''
+    ${userPostPatch}
```

```suggestion
    ${args.postPatch or ""}
```

I think that should work.